### ISSUE TITLE
nvidia-libs-384.69_1 fails to symlink /usr/lib/libnvidia-ptxjitcompiler.so.1 and /usr/lib/libnvidia-fatbinaryloader.so.1 → some CUDA applications will exhibit "unknown errors"

### ISSUE THREAD

**Issue by datenwolf** [voidlinux/void-packages#7494]
The package nvidia-libs-384.69_1 fails to symlink /usr/lib/libnvidia-ptxjitcompiler.so.1 and /usr/lib/libnvidia-fatbinaryloader.so.1

The result is that a subset of CUDA applications, namely those which do not ship with preassembled GPU binary code will exhibit "unknown errors", because the PTX compiler fails to load which will result in a lot of nonsensical CUDA errors.

It just took me a couple of hours to figure out why the CUDA project I'm working on failed to launch after a system update.

Adding the symlinks

```
lrwxrwxrwx 1 root root       35 Aug 24 20:16 /usr/lib/libnvidia-fatbinaryloader.so.1 -> libnvidia-fatbinaryloader.so.384.69
lrwxrwxrwx 1 root root       34 Aug 24 20:16 /usr/lib/libnvidia-ptxjitcompiler.so.1 -> libnvidia-ptxjitcompiler.so.384.69

```

Solved the problem.

**Comment by Hoshpak** [comment]
Are you sure that the fatbinaryloader-symlink is necessary? I just tested the installation of the current nvidia driver in a VM and the symlinks for libnvidia-ptxjitcompiler are indeed missing in the template. However no symlinks are created for libnvidia-fatbinaryloader.so by the installer and it is also not listed by the .manifest file in the installer.

**Comment by datenwolf** [comment]
Yes, it's absolutely necessary. I've even wrote a MCVE test program, because I kept scratching my head about what was going on. The moment a program contains a CUDA kernel (either `__device__` or `__global__`) for which no compiled binary is cached (under `~/.nv`) it is attempted to compile it from the embedded PTX. Only if that's necessary `libnvidia-fatbinaryloader.so.1` and `libnvidia-ptxjitcompiler.so.1` are `dlopen()`-ed. And if `dlopen()` fails it will just fall through without doing anything, giving you CUDA errors on completely unrelated calls, like `cudaStreamCreate`.

After I had the MCVE (before I knew what was happening) I was able to `strace` it. Then to seal the deal I started in `gdb` put a breakpoint on dlopen and saw that it never tried to open a unversioned or full versioned shared object file.

Adding those symlinks solve this problem.

Also I want to point out that these links exist on my Gentoo installation, where the drivers were installed through portage.

**Comment by Hoshpak** [comment]
Thanks. I amended my pull request to add symlinks for libnvidia-fatbinaryloader.so.${version}. It seems that Debian is creating no symlink and Arch is creating one for libnvidia-fatbinaryloader.so but not libnvidia-fatbinaryloader.so.1. Following the pattern of the other libraries in the package, I opted to create both symlinks.
